src/options.jl

```diff
@@ -0,0 +1,121 @@
+"""
+Options passed to `pgfplots` for various structures (`table`, `plot`, etc).
+
+Contents emitted in `key = value` form, or `key` when `value ≡ nothing`. Also
+see the [`@pgf`](@ref) convenience macro.
+"""
+const Options = OrderedDict
+
+function prockey(key)
+    if isa(key, Symbol) || isa(key, String)
+        return :($(string(key)) => nothing)
+    elseif @capture(key, (a_ : b_) | (a_ => b_) | (a_ = b_))
+        return :($(string(a))=>$b)
+    end
+    error("Invalid pgf option $key")
+end
+
+function procmap(d)
+  if @capture(d, f_(xs__))
+      return :($f($(map(procmap, xs)...)))
+  elseif !@capture(d, {xs__})
+      return d
+  else
+      return :($(Options)($(map(prockey, xs)...)))
+  end
+end
+
+"""
+    @pgf { ... }
+
+    @pgf some(nested(form({ ... })))
+
+Construct [`Options`](@ref) from comma-delimited `key` (without value),
+`key = value`, `key : value`, or `key => value` pairs enclosed in `{ ... }`,
+anywhere in the expression.
+
+Multi-word keys need to be quoted.
```

Comment:
Actually, you can use e.g. `only_marks` for this.

Reply:
Thanks, fixed.